testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/BrowserFlowTest.java

```diff
@@ -265,60 +505,74 @@ public void testSocialProvidersPresentOnLoginUsernameOnlyPageIfConfigured() {
-            AuthenticationExecutionModel execution3 = new AuthenticationExecutionModel();
-            execution3.setRequirement(AuthenticationExecutionModel.Requirement.ALTERNATIVE);
-            execution3.setAuthenticatorFlow(false);
-            execution3.setAuthenticator(OTPFormAuthenticatorFactory.PROVIDER_ID);
-            execution3.setPriority(30);
-            execution3.setParentFlow(formSubflow.getId());
-            execution3 = appRealm.addAuthenticatorExecution(execution3);
+            addAuthenticatorExecution(appRealm, AuthenticationExecutionModel.Requirement.ALTERNATIVE, OTPFormAuthenticatorFactory.PROVIDER_ID, 30, formSubflow);
 
             // Add OTPForm ALTERNATIVE execution
             appRealm.setBrowserFlow(newBrowserFlow);
-        });
+        };
+    }
+
+    private static AuthenticationFlowModel copyFlow(RealmModel appRealm, String original, String newFlowAlias) {
+        AuthenticationFlowModel existingBrowserFlow = appRealm.getFlowByAlias(DefaultAuthenticationFlows.BROWSER_FLOW);
+        return AuthenticationManagementResource.copyFlow(appRealm, existingBrowserFlow, newFlowAlias);
+    }
+
+    private static AuthenticationFlowModel prepareNewBrowserForms(RealmModel appRealm, AuthenticationFlowModel newBrowserFlow, String newFlowAlias) {
+        AuthenticationFlowModel formSubFlow = appRealm.getFlowByAlias(newFlowAlias + " forms");
+        List<AuthenticationExecutionModel> executions = appRealm.getAuthenticationExecutions(formSubFlow.getId());
+
+        // Remove all executions
+        for (AuthenticationExecutionModel authExecution : executions) {
+            appRealm.removeAuthenticatorExecution(authExecution);
+        }
+
+        return formSubFlow;
     }
 
+    private static void addAuthenticatorExecution(RealmModel realm, Requirement requirement, String providerId, int priority, AuthenticationFlowModel parentFlow) {
+        addAuthenticatorExecution(realm, requirement, providerId, priority, parentFlow, null);
+    }
+
+    private static void addAuthenticatorExecution(RealmModel realm, Requirement requirement, String providerId, int priority, AuthenticationFlowModel parentFlow, String configAlias) {
+        AuthenticationExecutionModel execution = new AuthenticationExecutionModel();
+        execution.setRequirement(requirement);
+        execution.setAuthenticatorFlow(false);
+        execution.setAuthenticator(providerId);
+        execution.setPriority(priority);
+        execution.setParentFlow(parentFlow.getId());
+        if (configAlias != null) {
+            execution.setAuthenticatorConfig(configAlias);
+        }
+        realm.addAuthenticatorExecution(execution);
+    }
+
+    private static void addAuthenticatorExecutionSubFlow(RealmModel realm, Requirement requirement, int priority, AuthenticationFlowModel subFlow, AuthenticationFlowModel parentFlow) {
+        AuthenticationExecutionModel execution = new AuthenticationExecutionModel();
```

Comment:
I think we can do a bit better here: have the method create the flow _and_ the executor.